src/components/utils/src/threads/thread_win.cc

```diff
@@ -121,7 +121,11 @@ Thread::Thread(const char* name, ThreadDelegate* delegate)
     , isThreadRunning_(false)
     , stopped_(false)
     , finalized_(false)
-    , thread_created_(false) {}
+    , thread_created_(false) {
+#ifdef THREAD_COUNT
+  ThreadCounter::Increment();
```

Comment:
@OHerasym do we have the common interface for all our threads? It seems like we can move this code to base's class ctor/dtor